core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -52,9 +54,9 @@
 
     private static ResourceReaper instance;
     private final DockerClient dockerClient;
-    private Map<String, String> registeredContainers = new ConcurrentHashMap<>();
-    private Set<String> registeredNetworks = Sets.newConcurrentHashSet();
-    private Set<String> registeredImages = Sets.newConcurrentHashSet();
+    private final Map<String, String> registeredContainers = new ConcurrentHashMap<>();
+    private final Set<String> registeredNetworks = synchronizedSet(new HashSet<>());
```

Comment:
`Sets.newConcurrentHashSet` wraps `ConcurrentHashMap` - we don't really benefit from its fine-grained locking